Output information:
```
gene_id<-gene id based on the annotation used
gene_symbol<-gene name based on the annotation used
transcript_id<-transcript id based on the annotation used
annotation<-gene biotype based on the annotation used
length<-length of the transcript
strand<-gene in the forward or reverse orientation
n_exons<-n of exons in the transcript
P_sites_sum<-Total sum of P-sites position mapped to the transcript
RNA_sites<-Total sum of RNA-sites position mapped to the transcript
Ribo_cov_aver<-Average coverage by all Ribo-seq reads on the transcript
RNA_cov_aver<-Average coverage by all RNA-seq reads on the transcript
category<-ORF category
ORF_id_tr<-ORF id with transcript coordinates
start_pos<-ORF start position (transcript coordinates)
stop_pos<-ORF stop position (transcript coordinates)
annotated_start<-Annotated ORF start position (transcript coordinates)
annotated_stop<-Annotated ORF stop position (transcript coordinates)
ORF_id_gen<-ORF id with genomic coordinates
ORF_length<-ORF length
ORF_P_sites<-Total sum of P-sites position mapped to the ORF
ORF_Psit_pct_in_frame<-Percentage of P-sites position mapped in frame
ORF_RNA_sites<-Total sum of RNA-sites position mapped to the ORF
ORF_RNAsit_pct_in_frame<-Percentage of RNA-sites position mapped in frame
ORF_pval_multi_ribo<-P-value for the multitaper test on the ORF region using P-sites positions
ORF_pval_multi_rna<-P-value for the multitaper test on the ORF region using RNA-sites positions
ORF_id_tr_annotated<-ORF id with annotated transcript coordinates
n_exons_ORF<-n of exons in the ORF
pct_covered_onlymulti_ribo<-percentage of the ORF region only supported by multimapping Ribo-seq reads
pct_covered_onlymulti_rna<-percentage of the ORF region only supported by multimapping RNA-seq reads
header_tofasta<-header used in the generated protein fasta file
ORF_pept<-peptide sequence of the identified ORF
```
